src/QueryM/Ogmios.purs

```diff
@@ -432,6 +447,248 @@ instance DecodeAeson TxEvaluationR where
   decodeAeson _ = Left
     (TypeMismatch "DecodeAeson TxEvaluationR is not implemented")
 
+---------------- PROTOCOL PARAMETERS QUERY RESPONSE & PARSING
+
+-- | A version of `Rational` with Aeson instance that decodes from `x/y`
+-- | representation, instead of `{ numerator, denominator }`
+newtype PParamRational = PParamRational Rational
+
+derive instance Newtype PParamRational _
+derive instance Generic PParamRational _
+
+instance Show PParamRational where
+  show = genericShow
+
+instance DecodeAeson PParamRational where
+  decodeAeson =
+    caseAesonString (Left err)
+      \string -> do
+        case String.split (Pattern "/") string of
+          [ numeratorStr, denominatorStr ] -> note err do
+            numerator <- BigInt.fromString numeratorStr
+            denominator <- BigInt.fromString denominatorStr
+            PParamRational <$> numerator % denominator
+          _ -> Left err
+    where
+    err = TypeMismatch "PParamRaional"
+
+type ProtocolParameters =
```

Comment:
I know that this type is easier for us to decode, but I'm not sure if it's the most useful representation for users. If they're expecting something like Plutus' `ProtocolParameters`, returning this type will at the very least oblige them to read the source code or the Ogmios documentation

Another consideration is that Haskell developers may not know that arbitrary strings can be used as record fields in PS (and may not know how to access the fields in that case)

We can keep the easier-to-parse structure, and then convert it in a function that uses field names that correspond more closely to what Plutus gives us. This could also let us adjust some of the types of each field (for example a normal `Rational` instead of `PParamRational` and `Coin`s instead of `BigInt`s in some places). I know that requires a non-trivial amount of boilerplate, but I think it's better that we shoulder such costs instead of users